content/projects/gsoc/mentors.adoc

```diff
@@ -47,18 +45,281 @@ Mentors are expected to...
+
+When we rank projects, we make sure mentors only get the number of projects they want (usually one or two)
+regardless of how many projects the mentor registers for.
+We also ask mentors to rank the projects they'd like to mentor in order of their preference.
+Org Admins make sure there is at least TWO mentors per project.
+The Org Admins help organize mentor teams and projects are ranked in a manner that tries to maximize success and happiness for everyone.
+
+Note that we allow mentors to participate in more than one project only if the mentor agrees with it.
+We do not recommend taking on more than two projects.
+You can still contribute in the Jenkins project while you mentor in GSoC.
+
+This all sounds complicated, but long story short, this phase is when the match making process formally happens between mentors, students and projects.
+
+The max number should never exceed our total mentoring capacity.
+The min number is the quantity of projects we feel confident we can mentor and succeed.
+For example We can feel very confident about 5 projects,
+and reasonably about 2 projects, and not enough confident about the rest.
+Then our min and max would be 5 and 7.
+
+Then we send our slot request min and max numbers to Google.
+
+==== During project selection phase
+
+*This phase lasts one week.*
+
+This phase start immediately when Google sends us our final number of slots. We may get only the minimum.
+Sometimes heart wrenching decisions need to be made at this time.
+
+During this phase, mentors and Jenkins Org Admins hold a private meeting to make the final project selection
+and the mentor teams are finalized and confirmed. Then we submit our final selection to the GSoC program.
+
+=== During the final selection awaiting period
+
+*This phase lasts a few days.*
+
+We wait. We are NOT allowed to communicate any information regarding the selection to students.
+
+=== During Community Bonding
+
+*This phase lasts 3 weeks.*
+
+*Expectation:* about 5 to 8 hours per week.
+
+This is the most critical phase when it comes to working with your student.
+Year after year, if this phase goes well, the rest of the program usually goes well, but if this phase does not go well, the project usually fails.
+
+Do not waste any time.
+Organize your first official meeting with the student, within the first week. Meet every week immediately.
+We have written a guide for this phase, link:../students/#community-bonding[read it] and follow it.
+
+In rare cases, it is acceptable to re-work the project idea, even change it completely, as long as the student and the mentors all agree.
+Written documentation about this is essential, if it happens, let the Org Admins know.
+
+Let Jenkins Org Admins know as soon as possible if there is any need for knowledge transfer sessions.
+We often organize special public presentations to go over plugin and core development flows and code that
+students and mentors alike benefit from attending.
+
```

Comment:
We should probably make sure at least the first coding phase has stories in Jira before the student starts coding

Reply:
Good suggestion. I added it to community bonding and eval periods.